doc/permissions.md

```diff
@@ -0,0 +1,53 @@
+# Permissions
```

Comment:
# Permissions

A permission in aurelia-acl is represented using strings arrays and objects.
The idea is to have a flexible way of granting, revoking and checking for
permissions. Aurelia-acl support *nested* permissions. Meaning one can put
objects or strings in arrays and arrays in object's properties.\

## Object
Internally aurelia-acl will normalize objects. Meaning that truthy values
become true and otherwise false. It also converts arrays and strings to
objects with boolean values.

```js
let userPermissions = {
  messages: {
    read: true,
    write: true,
    delete: false
  }
};
```
## Array
```js
let userPermissions = {
  messages: ['read', 'write', 'delete']
};
```

## String
If nesting is not your thing and you would like to keep things really flat and
simple, you should just use strings. `acl.permit('messages')` to grant messages
permissions.

```js
let userPermissions = 'messages';
```